### PR TITLE
Update registration page

### PR DESCRIPTION
**Pull request by puqeko**
* Expand detail on graduation date and eligibility
* Reformat to use headers instead of paragraphs
* <strike>Make "new school" slightly more noticeable</strike>
* Improve spacing of header for small devices

Avoided changing any stylesheets for fear of impacting other pages

<strike>Merge after PR #125 </strike>

### CODE REVIEW

**Comment by puqeko** [comment]
Screen Shot 2021-02-12 at 9 05 25 PM

File: app/views/accounts/registrations/new.html.erb
Comment (Holmes98) [inline app/views/accounts/registrations/new.html.erb:60]: Move the reCAPTCHA below "Submit" instead of "School".
```suggestion
  <h3>Submit</h3>
  <p>
    <%= recaptcha_tags :site_key => @db_settings["recaptcha/public_key"] if @db_settings["recaptcha/public_key"] && !@db_settings["recaptcha/public_key"].empty? %>
  </p>
```

File: app/views/accounts/registrations/new.html.erb
Comment (Holmes98) [inline app/views/accounts/registrations/new.html.erb:39]:
```suggestion
    <li>Anyone can compete in the NZIC unofficially without specifying a graduation date. However, you will not be ranked on official scoreboards or be eligible to be invited to our programming camps.</li>
    <li>Only New Zealand citizens or permanent residents that have attended camp will be considered to represent New Zealand at the International Olympiad in Informatics.</li>
```

**Review by Holmes98 — CHANGES_REQUESTED** [review]
Do we also want to change the "Edit Profile" page? (`app/views/user/_form.html.erb`)

File: app/views/accounts/registrations/new.html.erb
Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:53]: Okay. The black hover is indeed disgusting. I have never liked the grey links though

File: app/views/accounts/registrations/new.html.erb
Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:60]: Good spot

File: app/views/accounts/registrations/new.html.erb
Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:39]: Hmmm. I need to start typing things in something that does spell checking

File: app/views/accounts/registrations/new.html.erb
Comment (Holmes98) [inline app/views/accounts/registrations/new.html.erb:34]: Removing this causes new schools to be added with a nil `country_code`, so they won't appear under forms which use `School.where(country_code: "NZ")`.

Comment (Holmes98) [reply app/views/accounts/registrations/new.html.erb:34]: Should be fixed but someone else should probably review this.

Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:34]: Hmmm. Good spot. A terrible way to solve this could be to add it back in but make it a hidden element on the page.

Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:34]: If all schools are now only NZ schools then perhaps it is worth finding and replacing all instances of `School.where(country_code: "NZ")` and creating a db migration to drop this information. If so, I'd prefer to make this a separate pull request and reintroduce the field here so that this one doesn't get held up

Comment (Holmes98) [reply app/views/accounts/registrations/new.html.erb:34]: Sorry for not being clearer, but I already added a fix in c07c9f6. I just thought someone else should review my changes before we merge. c1378fa can be reverted.

Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:34]: Oh, okay, sorry. In that case, do we also need to change the `school` method in the `User` class too?

Comment (puqeko) [reply app/views/accounts/registrations/new.html.erb:34]: Clarification from @Holmes98 chat:

> If we remove the country code select box, POST requests will no longer contain the `user[school][country_code]` parameter (so it's treated as `nil`). Previously, this was used in the `school=` method to query for a school with that country code (the `first_or_initialize` part adds a new school if none was found). The fix is to just always use 'NZ' instead of that parameter.

**Review by Holmes98 — APPROVED** [review]

**Pull request merged**
